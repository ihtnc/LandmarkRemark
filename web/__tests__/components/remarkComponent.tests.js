import React from 'react';
import { shallow } from 'enzyme';
import Remark from "@components/remarkComponent";

describe("RemarkComponent", () => {
  let data = {};

  beforeEach(() => {
    data = {
      remarkId: 'remarkId',
      email: 'email@address',
      latitude: 123,
      longitude: 789,
      remark: 'remark'
    };
  });

  test("should render correctly when editable", () => {
    const wrapper = shallow(<Remark data={data} editable={true} />)
    expect(wrapper).toMatchSnapshot();
  });

  test("should render correctly when new", () => {
    const wrapper = shallow(<Remark data={data} editable={true} new={true} />)
    expect(wrapper).toMatchSnapshot();
  });

  test("should render correctly when not editable", () => {
    const wrapper = shallow(<Remark data={data} />)
    expect(wrapper).toMatchSnapshot();
  });
});