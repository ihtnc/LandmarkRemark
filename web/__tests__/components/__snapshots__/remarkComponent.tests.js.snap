// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemarkComponent should render correctly when editable 1`] = `
<styled.div>
  <styled.div>
    email@address
    :
  </styled.div>
  <styled.input
    onChange={[Function]}
    value="remark"
  />
  <styled.span
    error={false}
    show={false}
  />
  <styled.div>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Save
    </styled.button>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Delete
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`RemarkComponent should render correctly when new 1`] = `
<styled.div>
  <styled.div>
    email@address
    :
  </styled.div>
  <styled.input
    onChange={[Function]}
    value="remark"
  />
  <styled.span
    error={false}
    show={false}
  />
  <styled.div>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Save
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`RemarkComponent should render correctly when not editable 1`] = `
<styled.div>
  <styled.div>
    email@address
    :
  </styled.div>
  <styled.div>
    remark
  </styled.div>
  <styled.span
    error={false}
    show={false}
  />
</styled.div>
`;
